cpp/core.cc

```diff
@@ -20,6 +20,10 @@
 
 #include "_gen/frontend/consts.h"  // gVersion
 
+#ifndef HAVE_ENVIRON // handle systems that don't include environ in unistd
+extern char **environ;
```

Comment:
It is weird that OS X doesn't have this

I think it should be OK to unconditionally declare `extern char **environ`

Reply:
Okay, that let's us remove the check for its definition as well. As a note, it does appear that the `configure-coreutils` file included in this repo as testdata ends up setting a [`HAVE_ENVIRON_DECL` variable in it](https://github.com/oilshell/oil/blob/master/benchmarks/testdata/configure-coreutils#L10315) but I think you're right that setting an external var there should do no harm.